test/test_torch.py

```diff
@@ -10878,6 +10878,27 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=False) +
+              torch.testing.get_all_complex_dtypes()))
+    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_half=True) +
+                    torch.testing.get_all_complex_dtypes()))
+    def test_exp(self, device, dtype):
+        for v in (2, -2) + ((1j, 1 + 1j) if dtype.is_complex else ()):
```

Comment:
I think we should just generate `randn` input to get a more interesting input value and also since we are testing against `numpy`, we don't have to worry about defining the expected output so we can generate random values.

also it might be worth adding test for integers as well (if it doesn't already exist).

cc. @mruberry

Reply:
I also would prefer to see randn used, but we already have tests in TestTorchMathOps and TestTensorDeviceOps. I'm not sure what the value of this additional test is?

Reply:
I wasn't aware of `TestTorchMathOps`. I have now changed the PR to use `TestTorchMathOps`

Reply:
After a lot of fiddling, it looks like the current setup of `TestTorchMathOps` isn't quite ready for complex yet (indeed, I don't see other complex operators tested there). I'm reverting to this independent testing function, and I would propose to move this along with many other complex operators to use `TestTorchMathOps`. Is this OK?

Reply:
@xuhdev what's the issue with having a unified test? If there are any general things we need to add/fix to make the unified testing easier and feasible, I'd like to take a note of it and address it

Reply:
The current unified test simply doesn't have any thing enabled with complex numbers, and enabling `_complex_types` leads to a lot of errors related to `NaN`. Also, seems that besides this operator `exp`, other complex number operators do not have any tests yet. So... maybe a lot of unexpected work to do.

Reply:
What errors do you see? 
cc @mruberry

Reply:
@xuhdev ?

Reply:
For some reason I can't reproduce on my system, but I also didn't record the previous CI errors. I'll try to make another PR to demonstrate.

Reply:
I locally built and added test in `_TorchMathTestMeta` it's failing for a complex number with large real value and 0 imag value. 
```
>>>x=torch.tensor(9999999995.0000+0.0000j)
>>> torch.exp(x)
tensor((inf+nanj))
>>> np.exp(x.numpy())
__main__:1: RuntimeWarning: overflow encountered in exp
array([inf+0.j])
```
we should fix this

Reply:
@anjali411 Yes, we should fix this. Could you reproduce it on CUDA as well? The test that has broken is the CPU version, not CUDA.

Reply:
@anjali411 I've modified the test to note the erroneous behavior on CPU. I believe it is better to be fixed in a separate PR (this PR intends to add CUDA support and has already compounded a lot of test refactoring). Could we create an issue for it? I'm happy to look into it if I can find out the reason.

Reply:
For CUDA:
These cases look fine:
```
>>> x=float('inf')
>>> y=torch.tensor(x, dtype=torch.cfloat).cuda()
>>> y.exp()
tensor((inf+0.0000j), device='cuda:0')
>>> y=torch.tensor(x)*(1+1j)
>>> y=y.cuda()
>>> import numpy as np
>>> np.exp(y.cpu().numpy())
__main__:1: RuntimeWarning: invalid value encountered in exp
(inf+nanj)
>>> y
tensor((inf+infj), device='cuda:0')
>>> y.cpu()
tensor((inf+infj))
>>> torch.exp(y)
tensor((inf+nanj), device='cuda:0')
```

For completeness, can you add tests for these special cases (inf, 0), (0, inf), (inf, inf), (inf, nan), (nan, nan) etc.

cc. @mruberry

Reply:
I've found something puzzling. For the case `(0, inf)`, seems like I would never be able to do it in Python:

    >>> 0 + float('inf') * 1j
    (nan + infj)

any idea?

Reply:
I've added tests to cover all mentioned special cases.

Reply:
added some comments but looks good overall. once the comments are resolved and the flake8 warnings are gone and CI is green, it's good to go

Reply:
I think @anjali411 updated TestTorchMatchOps to test complex dtypes, too.